stream/rhcos/rhcos.go

```diff
@@ -16,3 +19,46 @@ type AzureDisk struct {
 	// from this URL as that may lead to performance limitations.
 	URL string `json:"url,omitempty"`
 }
+
+// Marketplace contains marketplace images for all clouds.
+type Marketplace struct {
+	Azure *AzureMarketplace `json:"azure,omitempty"`
+}
+
+// AzureMarketplaceImages contains both the HyperV- Gen1 & Gen2
+// images for a purchase plan.
+type AzureMarketplaceImages struct {
+	Gen1 *AzureMarketplaceImage `json:"hyperVGen1,omitempty"`
+	Gen2 *AzureMarketplaceImage `json:"hyperVGen2,omitempty"`
+}
+
+// AzureMarketplace lists images, both paid and
+// unpaid, available in the Azure marketplace.
+type AzureMarketplace struct {
+	// NoPurchasePlan is the standard, unpaid RHCOS image.
+	NoPurchasePlan *AzureMarketplaceImages `json:"no-purchase-plan,omitempty"`
+
+	// OCP is the paid marketplace image for OpenShift Container Platform.
+	OCP *AzureMarketplaceImages `json:"ocp,omitempty"`
+
+	// OPP is the paid marketplace image for OpenShift Platform Plus.
+	OPP *AzureMarketplaceImages `json:"opp,omitempty"`
+
+	// OKE is the paid marketplace image for OpenShift Kubernetes Engine.
+	OKE *AzureMarketplaceImages `json:"oke,omitempty"`
+}
```

Comment:
```suggestion
type AzureMarketplace map[string]AzureMarketplaceImages
```

Reply:
Hm, on further thought, it seems like it may be better to explicitly list the offers, rather than using an arbitrary string as the key (my suggestion). I have updated the list to now be complete by including the EMEA counterparts. 

Feedback is welcome. The offers have not changed since they were initially created several years ago.